dvc/repo/run.py

```diff
@@ -66,6 +70,35 @@ def run(
 
     with self.state:
         if not no_exec:
+            if (stage.cmd or stage.is_import) and not stage.locked:
+                existing = [
```

Comment:
looks like this list is calculated in a different way than in `stage.remove_outs`

Reply:
we also calculate this list no matter what, even if `--remove-outs` is specified. I'm not sure how expensive the changed_cache could be. Or can become in the future.

Reply:
Was the intention to show only outputs that are not in cache? Then the message is confusing. Because it actually should show _all_ outputs (not persisted) the way it's written.

Reply:
Agree, @shcheklein , message is misleading, the intention was to show it only when the output already exists and is not cached, but the latter doesn't make a lot of sense (in what cases would the output be cached? just when the output is already on another DVC-file).

I'll remove the `out.changed_cache` condition.